compiler/resolution/lowerIterators.cpp

```diff
@@ -2174,6 +2174,40 @@ static void fixNumericalGetMemberPrims()
 }
 
 
+//
+// For _toLeader fns, convert out-of-scope references to prim calls.
+// cleanupLeaderFollowerIteratorCalls() wouldn't handle such references
+// if a _toLeader call is inlined.
+//
+static void cleanupLeaderIteratorCalls()
+{
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    if (fn->inTree() && !strcmp(fn->name, "_toLeader")) {
+      ArgSymbol* toLeaderArg = fn->getFormal(1);
+      if (! toLeaderArg->type->symbol->hasFlag(FLAG_ITERATOR_RECORD))
+        continue;
+
+      std::vector<SymExpr*> symExprs;
+      collectSymExprs(fn, symExprs);
+      for_vector(SymExpr, se, symExprs) {
+        ArgSymbol* iteratorArg = toArgSymbol(se->symbol());
+        if (iteratorArg == NULL) continue;
+        if (iteratorArg->defPoint->parentSymbol == fn) continue;
+
+        SET_LINENO(se);
+        VarSymbol* ftemp = newTemp("ftemp", iteratorArg->qualType());
+        PrimitiveTag irf = PRIM_ITERATOR_RECORD_FIELD_VALUE_BY_FORMAL;
+        Expr*       stmt = se->getStmtExpr();
+        stmt->insertBefore(new DefExpr(ftemp));
+        stmt->insertBefore(new CallExpr(PRIM_MOVE, ftemp,
+                            new CallExpr(irf, toLeaderArg, iteratorArg)));
+        se->replace(new SymExpr(ftemp));
+      }
+    }
+  }
+}
```

Comment:
The PR message says this processing moved from cleanupLeaderFollowerIteratorCalls but I don't see a change to that function. Should there be one? Could both functions share some logic? I would have generally expected you to factor out the key parts of cleanupLeaderFollowerIteratorCalls into a function and call that here to do the transformation sooner. Is there a reason that doesn't work?